grammar.js

```diff
@@ -808,7 +801,7 @@ module.exports = grammar({
     parameter: $ => seq(
       optional($.uncurry),
       choice(
-        $._pattern,
+        seq($._pattern, optional($.type_annotation)),
```

Comment:
I removed `$.type_annotation` from `_pattern` rule. So it is necessary to add it here.